### PR TITLE
[tensorflow][build] Added apt upgrades to TF1.15-cpu,gpu,inf,train

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Adding APT Upgrade helps us consume security patches while rebuilding images and thus pass the sanity tests.

### DLC image/dockerfile

1. tensorflow/inference/docker/1.15/py36/Dockerfile.cpu
2. tensorflow/inference/docker/1.15/py36/cu100/Dockerfile.gpu
3. tensorflow/training/docker/1.15/py3/Dockerfile.cpu
4. tensorflow/training/docker/1.15/py3/cu100/Dockerfile.gpu
5. tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true` or `neuron_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by tejaschumbalkar — COMMENTED** [review]
Lets change dockerfiles in `py3` folder for TF1.15 Training images

File: tensorflow/inference/docker/1.15/py36/cu100/Dockerfile.gpu [hunk 1]
```diff
@@ -31,6 +31,7 @@ ENV MODEL_NAME=model
 ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update \
+ && apt-get -y upgrade \
```
Comment (shantanutrip) [inline tensorflow/inference/docker/1.15/py36/cu100/Dockerfile.gpu:34]: @Qingzi-Lan here

File: tensorflow/training/docker/1.15/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -31,6 +31,7 @@ ARG PIP=pip3
 ARG PYTHON_VERSION=3.7.10
 
 RUN apt-get update \
+ && apt-get -y upgrade \
```
Comment (shantanutrip) [inline tensorflow/training/docker/1.15/py3/Dockerfile.cpu:34]: here

File: tensorflow/training/docker/1.15/py3/cu100/Dockerfile.gpu [hunk 3]
```diff
@@ -28,6 +28,7 @@ ARG PIP=pip3
 ARG PYTHON_VERSION=3.7.10
 
 RUN apt-get update \
+ && apt-get -y upgrade \
```
Comment (shantanutrip) [inline tensorflow/training/docker/1.15/py3/cu100/Dockerfile.gpu:31]: here

File: tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu [hunk 4]
```diff
@@ -34,7 +34,7 @@ ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/est
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
 ARG SMDEBUG_VERSION=0.9.4
 
-RUN apt-get update && apt-get install -y --no-install-recommends --allow-unauthenticated \
+RUN apt-get update && apt-get -y upgrade && apt-get install -y --no-install-recommends --allow-unauthenticated \
```
Comment (shantanutrip) [inline tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu:37]: here

File: tensorflow/training/docker/1.15/py36/Dockerfile.cpu [hunk 5]
```diff
@@ -35,6 +35,7 @@ ARG PIP=pip3
 ARG PYTHON_VERSION=3.6.13
 
 RUN apt-get update \
+ && apt-get -y upgrade \
```
Comment (shantanutrip) [inline tensorflow/training/docker/1.15/py36/Dockerfile.cpu:38]: Change to ` && apt-get -y upgrade --only-upgrade systemd \`

File: tensorflow/training/docker/1.15/py36/cu100/Dockerfile.gpu [hunk 6]
```diff
@@ -32,6 +32,7 @@ ARG PIP=pip3
 ARG PYTHON_VERSION=3.6.13
 
 RUN apt-get update \
+ && apt-get -y upgrade \
```
Comment (shantanutrip) [inline tensorflow/training/docker/1.15/py36/cu100/Dockerfile.gpu:35]: Change to && apt-get -y upgrade --only-upgrade systemd \

File: tensorflow/inference/docker/1.15/py36/Dockerfile.cpu [hunk 7]
```diff
@@ -32,6 +32,7 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 # nginx + njs
 RUN apt-get update \
+ && apt-get -y upgrade \
```
Comment (shantanutrip) [inline tensorflow/inference/docker/1.15/py36/Dockerfile.cpu:35]: Change to && `apt-get -y upgrade --only-upgrade systemd \`

File: tensorflow/training/docker/1.15/py3/Dockerfile.cpu (hunk 2 above)
Comment (shantanutrip) [reply tensorflow/training/docker/1.15/py3/Dockerfile.cpu:34]: Change to `&& apt-get -y upgrade --only-upgrade systemd \`

File: tensorflow/training/docker/1.15/py3/cu100/Dockerfile.gpu (hunk 3 above)
Comment (shantanutrip) [reply tensorflow/training/docker/1.15/py3/cu100/Dockerfile.gpu:31]: Change to `&& apt-get -y upgrade --only-upgrade systemd \`

File: tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu (hunk 4 above)
Comment (shantanutrip) [reply tensorflow/training/docker/1.15/py3/cu110/Dockerfile.gpu:37]: Change to `&& apt-get -y upgrade --only-upgrade systemd \`

**Pull request closed without merging**